util/check_tool_requirements.py

```diff
@@ -223,6 +223,26 @@ def to_semver(self, version, from_req):
         return '.'.join(m.group(1, 2, 3))
 
 
+class VivadoToolReq(ToolReq):
+    tool_cmd = ['vivado', '-version']
+    version_regex = re.compile(r'Vivado v(.*)\s')
+
+    def to_semver(self, version, from_req):
+        # Regular Vivado releases just have a major and minor version.
+        # In this case, we set the patch level to 0.
+        m = re.fullmatch(r'([0-9]+)\.([0-9]+)\.*([0-9]*)', version)
+        if m is None:
+            raise ValueError("{} has invalid version string format."
+                             .format(version))
+
+        if m.group(3):
+            version = (m.group(1, 2, 3))
+        else:
+            version = (m.group(1), m.group(2), '0')
```

Comment:
Sorry, I should have said this earlier. This would probably be neater as something like `(m.group(1), m.group(2), m.group(3) or '0')`

Reply:
Much more elegant!